targets/targets.json

```diff
@@ -34,6 +34,10 @@
             "mpu-rom-end": {
                 "help": "Last address of ROM protected by the MPU",
                 "value": "0x0fffffff"
+            },
+            "default-form-factor": {
```

Comment:
Hm you are not setting default value here, but in the code sort of circumventing the config system and hiding the fact. Wouldn't it be cleaner to set it here?

Reply:
Setting the default here to `arduino` would also be misleading since some boards are not in the arduino form factor. Because of this @jeromecoutant [suggested keeping this null and specifying it at the target level](https://github.com/ARMmbed/mbed-os/pull/9449#discussion_r249683608), which I think is a good idea.

Reply:
Adding this new config is OK.
But who is going to replace the supported_form_factors line by this new config?
Why you are mixing both configurations ?

Reply:
Hi
No answer for this question?
A new config is created, but never used...!

Reply:
Hi @jeromecoutant sorry for the late response, I just noticed your question. The `default-form-factor` is intended to be used with `supported_form_factors`, not to replace it. Some boards may have multiple form factors that are testable and this value is intended to let you pick the one to test with. This is currently not needed by any targets so it is not used. I added it here for completeness. If you are concerned with it I can remove it an submit it as a PR only when it is needed. Let me know if you have a preference on this.

Reply:
@jeromecoutant even though this PR has been merged please provide any feedback you may have. I'll address any changes needed in a separate PR.